models/attachment.go

```diff
@@ -75,29 +81,32 @@ func (a *Attachment) DownloadURL() string {
 func NewAttachment(attach *Attachment, buf []byte, file io.Reader) (_ *Attachment, err error) {
 	attach.UUID = gouuid.NewV4().String()
 
-	localPath := attach.LocalPath()
-	if err = os.MkdirAll(path.Dir(localPath), os.ModePerm); err != nil {
-		return nil, fmt.Errorf("MkdirAll: %v", err)
+	fs := storage.FileStorage{
+		Ctx:      context.Background(),
+		Path:     setting.AttachmentPath,
+		FileName: attach.AttachmentBasePath(),
 	}
 
-	fw, err := os.Create(localPath)
+	fw, err := fs.NewWriter()
 	if err != nil {
 		return nil, fmt.Errorf("Create: %v", err)
 	}
-	defer fw.Close()
 
```

Comment:
just keep `defer fw.Close()` is better.

Reply:
`defer fw.Close()` can't be used here.
`fw` must be closed before `fi, err := fs.Attributes()` (Line 105). It can't open the file for attributes unless the `fw.Close()` is called.

That's why I had to do like this.

Reply:
Sorry for the late. 
Your suggestion will work fine here and will do that.